backend/routers/v1/holdings.py

```
@@ -0,0 +1,106 @@
+    ).all()
+    return holdings
+
+
+@router.get(
+    "/{portfolio_id}/{stock_id}",
+    response_model=db.schema.HoldingRead
+)
+def read_holding(
+    portfolio_id: str,
+    stock_id: str,
+    session: Session = Depends(get_session)
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)